cmd/otel-allocator/internal/allocation/allocator.go

```diff
@@ -25,19 +29,48 @@ import (
 
 var _ Allocator = &allocator{}
 
-func newAllocator(log logr.Logger, strategy Strategy, opts ...Option) Allocator {
+func newAllocator(log logr.Logger, strategy Strategy, opts ...Option) (Allocator, error) {
+	meter := otel.GetMeterProvider().Meter("targetallocator")
+	// targetsPerCollector records how many targets have been assigned to each collector.
+	// It is currently the responsibility of the strategy to track this information.
+	targetsPerCollector, err := meter.Int64Gauge("opentelemetry_allocator_targets_per_collector", metric.WithDescription("The number of targets for each collector."))
+	if err != nil {
+		return nil, err
+	}
+	collectorsAllocatable, err := meter.Int64Gauge("opentelemetry_allocator_collectors_allocatable", metric.WithDescription("Number of collectors the allocator is able to allocate to."))
+	if err != nil {
+		return nil, err
+	}
+	timeToAssign, err := meter.Float64Histogram("opentelemetry_allocator_time_to_allocate", metric.WithDescription("The time it takes to allocate"))
+	if err != nil {
+		return nil, err
+	}
+	targetsRemaining, err := meter.Int64Gauge("opentelemetry_allocator_targets_remaining", metric.WithDescription("Number of targets kept after filtering."))
+	if err != nil {
+		return nil, err
+	}
+	targetsUnassigned, err := meter.Int64Gauge("opentelemetry_allocator_targets_unassigned", metric.WithDescription("Number of targets that could not be assigned due to missing node label."))
+	if err != nil {
+		return nil, err
+	}
+
 	chAllocator := &allocator{
 		strategy:                      strategy,
 		collectors:                    make(map[string]*Collector),
 		targetItems:                   make(map[target.ItemHash]*target.Item),
 		targetItemsPerJobPerCollector: make(map[string]map[string]map[target.ItemHash]bool),
 		log:                           log,
+		targetsPerCollector:           targetsPerCollector,
+		collectorsAllocatable:         collectorsAllocatable,
+		timeToAssign:                  timeToAssign,
+		targetsRemaining:              targetsRemaining,
+		targetsUnassigned:             targetsUnassigned,
```

Comment:
Is storing metrics as fields idiomatic with the Go Otel SDK?

Reply:
should be good then :D resolving